common/ta_errors.c

```diff
@@ -0,0 +1,158 @@
+      return "Invald request value in JSON\n";
+    case SC_CACHE_NULL:
+      return "NULL parameters in cache\n";
+    case SC_CACHE_FAILED_RESPONSE:
+      return "Failed in cache operations\n";
+    case SC_CACHE_OFF:
+      return "Cache server doesn't turn on\n";
+    case SC_MAM_OOM:
+      return "Failed to create MAM object\n";
+    case SC_MAM_NULL:
+      return "NULL object in MAM\n";
+    case SC_MAM_NOT_FOUND:
+      return "Empty result from MAM\n";
+    case SC_MAM_FAILED_INIT:
+      return "Error in MAM initialization\n";
+    case SC_MAM_FAILED_RESPONSE:
+      return "Error in MAM response\n";
+    case SC_MAM_FAILED_DESTROYED:
+      return "Error in MAM destroy\n";
+    case SC_MAM_NO_PAYLOAD:
+      return "No payload or no chid\n";
+    case SC_MAM_FAILED_WRITE:
+      return "Failed to write\n";
+    case SC_MAM_FILE_SAVE:
+      return "Failed to save MAM file\n";
+    case SC_MAM_ALL_MSS_KEYS_USED:
+      return "All MSS private keys of current given parameters are used\n";
+    case SC_MAM_FAILED_CREATE_OR_GET_ID:
+      return "Failed to created/get chid or epid or msg_id\n";
+    case SC_MAM_FAILED_WRITE_HEADER:
+      return "Failed to write on header\n";
+    case SC_RES_OOM:
+      return "Failed to create response object\n";
+    case SC_RES_NULL:
+      return "NULL object in response\n";
+    case SC_CONF_NULL:
+      return "NULL object in response\n";
+    case SC_CONF_MISSING_ARGUMENT:
+      return "No argument in CLI\n";
+    case SC_CONF_UNKNOWN_OPTION:
+      return "Undefined option in CLI\n";
+    case SC_CONF_LOCK_INIT:
+      return "Failed to init lock\n";
+    case SC_CONF_LOCK_DESTROY:
+      return "Failed to destroy lock\n";
+    case SC_CONF_PARSER_ERROR:
+      return "Failed to initialize yaml parser\n";
+    case SC_CONF_FOPEN_ERROR:
+      return "Failed to open file\n";
+    case SC_UTILS_NULL:
+      return "NULL object in utils\n";
+    case SC_UTILS_WRONG_REQUEST_OBJ:
+      return "Wrong TA request object\n";
+    case SC_UTILS_TIMER_ERROR:
+      return "Errors occurred in timer function\n";
+    case SC_UTILS_TIMER_EXPIRED:
+      return "Timer expired\n";
+    case SC_HTTP_OOM:
+      return "Failed to create http object\n";
+    case SC_HTTP_NULL:
+      return "NULL object in http\n";
+    case SC_HTTP_INVALID_REGEX:
+      return "Invalid URL regular expression rule in http\n";
+    case SC_HTTP_URL_NOT_MATCH:
+      return "URL doesn't match regular expression rule\n";
+    case SC_HTTP_URL_PARSE_ERROR:
+      return "URL parameter parsing error\n";
+    case SC_MQTT_OOM:
+      return "Failed to create MQTT object\n";
+    case SC_MQTT_NULL:
+      return "NULL object in MQTT\n";
+    case SC_MQTT_INVALID_REGEX:
+      return "Invalid URL regular expression rule in MQTT\n";
+    case SC_MQTT_TOPIC_NOT_MATCH:
+      return "Topic doesn't match regular expression rule\n";
+    case SC_MQTT_URL_PARSE_ERROR:
+      return "Topic parameter parsing error\n";
```

Comment:
It's URL parsing error, correct?

Reply:
Topic parsing in MQTT is moved to `connectivity/common.c` in other pull request. Since this error code is not used now, I'll remove it directly.